symcache/src/writer.rs

```diff
@@ -261,17 +261,17 @@ impl<W: Write> SymCacheWriter<W> {
             }
 
             if (*last_addr == !0 || sym_addr >= *last_addr) && sym_addr < cur_addr {
-                self.write_simple_function(sym_addr, sym_len, sym)?;
+                self.write_simple_function(sym_addr, symbol.len().unwrap_or(!0), symbol.name())?;
                 symbol_iter.next();
-                *last_addr = sym_addr + sym_len as u64;
+                *last_addr = sym_addr + symbol.len().unwrap_or(1);
```

Comment:
Previously, we added `!0u32 as u64` in case the length was missing. From what I have seen the **only** reason this could be happening should be the last symbol. Just to be safe, I fall back to adding `1u64` here instead. This advances to the next symbol in any case and prevents an endless loop.